From c7f9df7938a72421a5875451b4eb08ac0f78b04c Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Mon, 24 Apr 2023 09:03:09 -0500
Subject: [PATCH 460/508] rpmsg: char: Update local endpt address for
 virtio-rpmsg backend

The rpmsg char driver creates a local end-point when the actual
endpt device is opened. The virtio-rpmsg backend can dynamically allocate
the local end-point address if the endpt creation is done using the
address RPMSG_ADDR_ANY. This is not reflected in the sysfs src file, so
update the stored address with the allocated address in such a case. This
allows the userspace to be able to retrieve the local end-point
address through sysfs.

Signed-off-by: Suman Anna <s-anna@ti.com>
Signed-off-by: Devarsh Thakkar <devarsht@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 drivers/rpmsg/rpmsg_char.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/rpmsg/rpmsg_char.c b/drivers/rpmsg/rpmsg_char.c
index 3e0b8f3496ed..5ee22eb92e07 100644
--- a/drivers/rpmsg/rpmsg_char.c
+++ b/drivers/rpmsg/rpmsg_char.c
@@ -147,6 +147,9 @@ static int rpmsg_eptdev_open(struct inode *inode, struct file *filp)
 	}
 
 	eptdev->ept = ept;
+	if (eptdev->chinfo.src == RPMSG_ADDR_ANY)
+		eptdev->chinfo.src = ept->addr;
+
 	filp->private_data = eptdev;
 	mutex_unlock(&eptdev->ept_lock);
 
-- 
2.41.0

